### ISSUE TITLE
Type error with Typescript 4.8

### ISSUE THREAD

**Issue by fcollonval** [vega/vega-lite#8465]
The following template parameter is throwing an error with TypeScript 4.8 because the template of `Conditional` type is restricted to a subset `Conditional<CD extends FieldDef<any> | DatumDef | ValueDef<any> | ExprRef | SignalRef>`

https://github.com/vega/vega-lite/blob/c97e30a8da569aabab7f348673899f43aa3b8825/src/channeldef.ts#L158

The error raised is 

```
  ../node_modules/vega-lite/build/src/channeldef.d.ts(63,66): error TS2344: Type 'T' does not satisfy the constraint 'SignalRef | ExprRef | FieldDef<any, any> | DatumDef<string, SignalRef | DateTime | ExprRef | PrimitiveValue> | ValueDef<...>'.
    Type 'T' is not assignable to type 'DatumDef<string, SignalRef | DateTime | ExprRef | PrimitiveValue>'.
```

Seen in https://github.com/jupyterlab/jupyterlab/actions/runs/3217544851/jobs/5260639447

**Comment by fcollonval** [comment]
Yes it is - but using compiler option is not an option. The code really requires the fixed mentioned in #8454 log:

```
63 export declare function isConditionalParameter<T>(c: Conditional<T>): c is ConditionalParameter<T>;
                                                  ~
This type parameter might need an `extends SignalRef | ExprRef | FieldDef<any, any> | DatumDef<string, SignalRef | DateTime | ExprRef | PrimitiveValue> | ValueDef<...>` constraint.
```

The reason is the type constraint on `Conditional<T extends ...>` and `ConditionalParameter<T extends ...>`. So the type parameter in `isConditionalParameter` function must extend it too.

**Comment by domoritz** [comment]
I see. If `skipLibCheck` is not an option for you, do you have cycles to help with fixing the types in Vega-Lite? Does it make sense to track this under https://github.com/vega/vega-lite/issues/5477 instead of this separate issue?
